src/app.js

```diff
@@ -129,9 +129,24 @@ export default class App extends React.Component {
             return (
               <AppShell location={props.location} releaseBranch={releaseBranch}>
                 <Switch>
-                  {routes.map((route, routeIndex) => (
-                    <Route key={`route-${routeIndex}`} {...route} />
-                  ))}
+                  {routes.map((route, routeIndex) => {
+                    if (route.renderWithProps) {
+                      const render = props => (
+                        <route.renderWithProps
+                          releaseBranch={releaseBranch}
+                          {...props}
+                        />
+                      )
+                      return (
+                        <Route
+                          key={`route-${routeIndex}`}
+                          render={render}
+                          {...props}
+                        />
+                      )
+                    }
+                    return <Route key={`route-${routeIndex}`} {...route} />
+                  })}
```

Comment:
Is it possible to create `Route` simpler?  like

 
```
const routeProps = route.renderWithProps ? {
  render: props => <route.renderWithProps releaseBranch=. {releaseBranch} {...props}/>,
  ...props
} : route;
return <Route key={`route-${routeIndex}`} {...routeProps}/>;
```

Reply:
update fcbcdb4[https://github.com/twreporter/twreporter-react/pull/2122/commits/fcbcdb48cd542d67a3b70a533f02124f70017ee6]

Reply:
nice, thanks.